network/src/peers/peer/outbound_handler.rs

```diff
@@ -110,8 +111,9 @@ impl Peer {
                 })?;
 
                 metrics::increment_counter!(metrics::outbound::ALL_SUCCESSES);
-                metrics::decrement_gauge!(OUTBOUND, 1.0);
+
                 self.queued_outbound_message_count.fetch_sub(1, Ordering::SeqCst);
+                metrics::decrement_gauge!(OUTBOUND, 1.0);
```

Comment:
this ordering change is for better readability, so that it's more clear that this metric is tightly bound to an atomic